src/keys.rs

```diff
@@ -112,7 +116,7 @@ impl From<&SpendingKey> for SpendAuthorizingKey {
 /// $\mathsf{ak}$ but stored here as a RedPallas verification key.
 ///
 /// [orchardkeycomponents]: https://zips.z.cash/protocol/nu5.pdf#orchardkeycomponents
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialOrd, Ord)]
 pub struct SpendValidatingKey(redpallas::VerificationKey<SpendAuth>);
 
 impl From<&SpendAuthorizingKey> for SpendValidatingKey {
```

Comment:
This depends on the fact that `SpendAuthorizingKey` will result in a `SpendValidatingKey` with ỹ = 0 (so that `SpendValidatingKey::to_bytes` corresponds to `I2LEOSP<sub>256</sup>(ak)`.